site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,725 @@
+        public HttpResponse hello() { ... }
+    }
+
+    // ...
+
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyGlobalDecorator1());      // order 1
+
+
+.. _media_type_nego:
+
+Media type negotiation
+----------------------
+
+Armeria provides `@ProduceType`_ and `@ConsumeType`_ annotations to support media type negotiation. It is not
+necessary if you have only one service method for a path and an HTTP method. Assume that you have multiple
+service methods for the same path and the same HTTP method as follows.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        public HttpResponse hello1() {
+            // Return a text document to the client.
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Armeria");
+        }
+
+        @Get("/hello")
+        public HttpResponse hello2() {
+            // Return a JSON object to the client.
+            return HttpResponse.of(HttpStatus.OK, MediaType.JSON_UTF8, "{ \"name\": \"Armeria\" }");
+        }
+    }
+
+If media type is not specified on any methods bound to the same path pattern, the first method declared will
+be used and the other methods will be ignored. In this example, ``hello1()`` will be chosen and the client
+will always receive a text document. What should you do if you want to get a JSON object from the path
+``/hello``? You can just specify the type of the content your method produces as follows and add an
+``Accept`` header to your client request.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @ProduceType("text/plain")
+        public HttpResponse helloText() {
+            // Return a text document to the client.
+            return HttpResponse.of(HttpStatus.OK, MediaType.PLAIN_TEXT_UTF_8, "Armeria");
+        }
+
+        @Get("/hello")
+        @ProduceType("application/json")
+        public HttpResponse helloJson() {
+            // Return a JSON object to the client.
+            return HttpResponse.of(HttpStatus.OK, MediaType.JSON_UTF8, "{ \"name\": \"Armeria\" }");
+        }
+    }
+
+A request like the following would get a text document::
+
+    GET /hello HTTP/1.1
+    Accept: text/plain
+
+A request like the following would get a JSON object::
+
+    GET /hello HTTP/1.1
+    Accept: application/json
+
+.. note::
+
+    Note that ``Content-Type`` header of a response is not automatically set. You are able to get the
+    negotiated `@ProduceType`_ from ``ServiceRequestContext.negotiatedProduceType()`` method.
+
+If a client sends a request without ``Accept`` header (or sending ``Accept`` header with a unsupported content
+type), it would be usually mapped to ``helloJson()`` method because the methods are sorted by the name of
+the type in an alphabetical order.
+In this case, you can adjust the order of the methods with `@Order`_ annotation. The default value of
```

Comment:
How about inserting an empty line after `an alphabetical order.`? so that we get a paragraph break?